addons/account/models/account_move_line.py

```diff
@@ -791,7 +786,7 @@ def _compute_product_uom_id(self):
         for line in self:
             # vendor bills should have the product purchase UOM
             if line.move_id.is_purchase_document():
-                line.product_uom_id = line.product_id.uom_po_id
+                line.product_uom_id = line.product_id.seller_ids.filtered(lambda s: s.partner_id == line.partner_id).product_uom_id or line.product_id.uom_id
```

Comment:
This is risky, a product can have the same seller with different quantities/uoms/prices, so you may end up with the wrong UoM. Or I guess in this case, an error since you'll have 2+ `product_uom_id`s